core/src/smartcontracts/wasm.rs

```diff
@@ -767,26 +771,36 @@ impl<S: state::LimitsMut> Runtime<S> {
 
 #[allow(clippy::needless_pass_by_value)]
 impl<S: state::Authority + state::Wsv + state::WsvMut> Runtime<S> {
-    fn default_execute_query(query: QueryBox, state: &S) -> Result<Value, ValidationFail> {
-        iroha_logger::debug!(%query, "Executing");
+    fn default_execute_query(
+        query_request: SmartContractQueryRequest,
+        state: &S,
+    ) -> Result<BatchedResponse<Value>, ValidationFail> {
+        iroha_logger::debug!(%query_request, "Executing");
 
         let wsv = state.wsv();
 
-        // NOTE: Smart contract (not executor) is trying to execute the query, validate it first
-        // TODO: Validation should be skipped when executing smart contract.
-        // There should be two steps validation and execution. First smart contract
-        // is validated and then it's executed. Here it's validating in both steps.
-        // Add a flag indicating whether smart contract is being validated or executed
```

Comment:
isn't this comment still valid? should we make a ticket?

Reply:
or now we're validating while executing?

Reply:
I think it's outdated.

Basic smart contracts fallback to `default_execute_query()` (you can find this associated function in `core/src/wasm.rs`). This functions does validation by calling `wasm.executor().validate_query()` and then calls `query.execute()`.

But this is not the case for validator. You can check how functions are implemented for any runtime implementing `ExecuteOperationsAsExecutorMut`. Here `execute_query()` will just call `query.execute()` without validation